arduino/smartcar/smartcar.ino

```diff
@@ -136,3 +136,16 @@ void preventCrashWithUSandIR(DistanceSensor& distanceSensor) {
       delay (200);
     }
 }
+void detectSurface(DistanceSensor& distanceSensor){
+  unsigned int distance = distanceSensor.getDistance();
+  if (distance > minDistance && distance < obstacleAvoidDistance) {
+      Serial.print(distanceSensor.getDistance());
+      Serial.println(" Obstacle detected. ");
+      car.setSpeed(0);
+      delay(20000);
+      car.overrideMotorSpeed(50, -100);
+      delay (800);
+      car.setSpeed(0);
+      delay (20000);
+  }
```

Comment:
What's using this and how is this significantly different from `preventCrashWithUSandIR`?

Reply:
This was just a small adjustment and is not completely finished. I figured since you and Krasen prefered that we would make some smaller contributions for you to keep track of some work. The rest of this task will be finished today.

Reply:
> This was just a small adjustment and is not completely finished. I figured since you and Krasen prefered that we would make some smaller contributions for you to keep track of some work. The rest of this task will be finished today.

Just to clarify, we want smaller **functional** pieces of work. If something is not used it is by definition not functional.